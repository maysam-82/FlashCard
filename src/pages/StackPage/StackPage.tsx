import React from 'react';
import { connect } from 'react-redux';
import { IStoreState } from '../../redux/reducers/index';
import { IStack } from '../../redux/actions';
import Card from '../../components/Card';

export interface IProps {
	stack: IStack;
}

export const StackPage: React.SFC<IProps> = ({ stack }) => {
	const renderCards = () => {
		return (stack.cards || []).map(({ prompt, id, answer }) => (
			<Card key={id} prompt={prompt} answer={answer} />
		));
	};
	return (
		<div>
			<h4>{stack.title}</h4>
			<div>{renderCards()}</div>
		</div>
	);
};

const mapStateToProps = (state: IStoreState) => ({
	stack: state.stack,
});
export default connect(mapStateToProps)(StackPage);
