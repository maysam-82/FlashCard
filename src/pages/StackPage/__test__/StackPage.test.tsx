import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { StackPage } from '../StackPage';
import { testStack } from '../../../fixtures/testData/testData';

describe('StackPage component', () => {
	let component: ShallowWrapper;
	beforeEach(() => {
		component = shallow(<StackPage stack={testStack} />);
	});
	it('should render StackPage correctly', () => {
		expect(component).toMatchSnapshot();
	});
	it('should render 1 Card component', () => {
		expect(component.find('Cards').length).toEqual(1);
	});
	describe('When there is no `card` in `stack.cards`', () => {
		beforeEach(() => {
			component = shallow(<StackPage stack={testStack} />);
			component.setProps({ stack: { ...testStack, cards: [] } });
		});
		it('should render no Cards', () => {
			expect(component.find('Cards')).toHaveLength(0);
		});
	});
});
