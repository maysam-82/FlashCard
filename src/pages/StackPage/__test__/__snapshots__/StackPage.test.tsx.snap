// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackPage component should render StackPage correctly 1`] = `
<div>
  <h4>
    test title stack
  </h4>
  <Cards
    cards={
      Array [
        Object {
          "answer": "test answer",
          "id": 2,
          "prompt": "test prompt",
        },
      ]
    }
  />
</div>
`;
